entities/src/instance.rs

```diff
@@ -45,3 +45,34 @@ pub struct Stats {
     status_count: u64,
     domain_count: u64,
 }
+
+/// Rules of an instance
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct Rule {
+    /// An identifier for the rule.
+    pub id: String,
+    /// The rule to be followed.
+    pub text: String,
+}
+
+/// An instance-level domain block.
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct DomainBlock {
+    /// URI of the domain in question
+    pub domain: String,
+    /// Digest
+    pub digest: String,
+    /// Severity of the block
+    pub severity: String,
+    /// Admin's public comment.
+    pub comment: String,
+}
+
+/// Weekly activity on an instance
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct Activity {
+    pub week: String,
+    pub statuses: String,
+    pub logins: String,
+    pub registrations: String,
```

Comment:
I'm sorry that I didn't see the updates to this, it was showing up out of order and I thought this hadn't been done yet.

```suggestion
    /// UNIX Timestamp at midnight at the first day of the week.
    pub week: String,
    /// The number of Statuses created since the week began (cast from an integer)
    pub statuses: String,
    /// The number of user logins since the week began (cast from an integer)
    pub logins: String,
    /// The number of user registrations since the week began (cast from an integer)
    pub registrations: String,
```